creation/create_cvmfsexec_distros.sh

```diff
@@ -3,123 +3,196 @@
-        echo "Found newer version of cvmfsexec..."
-        echo "Rebuilding cvmfsexec distributions using the latest version..."
-    fi
 
-    # build the distributions for cvmfsexec based on the source, os and platform combination
-    cvmfsexec_distros="$cvmfsexec_temp"/distros
-    if [[ ! -d "$cvmfsexec_distros" ]]; then
-        mkdir -p "$cvmfsexec_distros"
-    fi
+####################### MAIN SCRIPT STARTS FROM HERE #######################
 
-    declare -a cvmfs_sources
-    cvmfs_sources=($(echo $CVMFS_SOURCES | tr ":" "\n"))
-
-    declare -a machine_types
-    machine_types=($(echo $SUPPORTED_TYPES | tr ":" "\n"))
-
-    for cvmfs_src in "${cvmfs_sources[@]}"
-    do
-        for mach_type in "${machine_types[@]}"
-        do
-            echo -n "Making $cvmfs_src distribution for $mach_type machine..."
-            os=`echo $mach_type | awk -F'-' '{print $1}'`
-            arch=`echo $mach_type | awk -F'-' '{print $2}'`
-            if "$cvmfsexec_latest"/makedist -m $mach_type $cvmfs_src &> /dev/null ; then
-                "$cvmfsexec_latest"/makedist -o "$cvmfsexec_distros"/cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-                if [[ -e "$cvmfsexec_distros"/cvmfsexec-${cvmfs_src}-${os}-${arch} ]]; then
-                    echo " Success"
-                    tar -cvzf "$cvmfsexec_tarballs"/cvmfsexec_${cvmfs_src}_${os}_${arch}.tar.gz -C "$cvmfsexec_distros" cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-                fi
-            else
-                echo " Failed! REASON: $cvmfs_src may not yet have a $mach_type build."
-            fi
-
-            # delete the dist directory within cvmfsexec to download the cvmfs configuration
-            # and repositories for another machine type
-            rm -rf "$cvmfsexec_latest"/dist
-        done
-    done
-
-    # remove the distros and latest folder under cvmfsexec.tmp
-    rm -rf "$cvmfsexec_distros"
-    rm -rf "$cvmfsexec_latest"
+# parsing the command-line arguments
+# check whether the first argument passed is the option '--work-dir'
+if [[ $1 == "--work-dir" && ! -d "$2" ]]; then
+	# if value after this option is not a path (it is empty/unspecified), invoke the error handler
+	error_handler "The value of --work-dir option must be an existing directory."
+elif [[ -d "$1" ]]; then
+	# check if there was a path specified without the --work-dir option, invoke the error handler
+	error_handler "--work-dir option should precede directory path."
+fi
+# otherwise, continue parsing the arguments
+if [[ $1 == "--work-dir" ]]; then
+	work_dir="$2"
+	shift 2
+else
+	# if --work-dir is not passed, assume default work-dir (RPM install)
+	work_dir="$DEFAULT_WORK_DIR"
 fi
 
+# check whether there are any remaining arguments passed to the script after the option
+if [[ $# -eq 0 ]]; then
+	echo -e "\nNo sources specified. Building/Rebuilding of cvmfsexec distributions disabled!"
```

Comment:
I see many `echo -e "\n..."`. It is OK not to add empty lines ahead of messages and previous messages should end w/ newline so this will start on its own line.

Reply:
This has been updated. The reason newline character was added to some of the `echo` statements was to be able to print the message on a newline, which was not happening without the newline character at the beginning of the message. To change this, the `factory_initd_startup_template` and `factory_initd_startup_template_sl7` scripts in `creation/template` were also modified.